src/API/Reports/Customers/DataStore.php

```diff
@@ -626,6 +625,34 @@ public static function get_customer_id_by_user_id( $user_id ) {
 		return $customer_id ? (int) $customer_id : false;
 	}
 
+	/**
+	 * Retrieve the last order made by a customer.
+	 *
+	 * @param int $customer_id Customer ID.
+	 * @return object Order|false.
+	 */
+	public static function get_last_order( $customer_id ) {
+		global $wpdb;
+		$orders_table                = $wpdb->prefix . 'wc_order_stats';
+		$excluded_statuses_condition = "AND status IN ( '" . implode( "','", array_map( 'esc_sql', array_keys( wc_get_order_statuses() ) ) ) . "' )";
+
+		$last_order = $wpdb->get_var(
+			$wpdb->prepare(
+				// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+				"SELECT order_id FROM {$orders_table}
+				WHERE customer_id = %d
+				{$excluded_statuses_condition}
+				ORDER BY order_id DESC LIMIT 1",
```

Comment:
Similar query to `get_oldest_orders` (below), we just DESC by order id, instead of ASC.

Reply:
Of course my test order failed this case - it is the newest order (as default sorted in the dashboard by date), but it doesn't have the highest numerical ID. Perhaps we should sort using `date_created` or `date_created_gmt` here?

Reply:
Makes sense, I will update it to sort by `date_created_gmt`. I worked off the original [get_oldest_orders](https://github.com/woocommerce/woocommerce/blob/13f429e80de7c479443fc407f0aab0eb1ac322ad/includes/data-stores/class-wc-customer-data-store.php#L327-L347) function that sorted by post id. But `date_created` indeed makes more sense.

Reply:
updated all changes in https://github.com/woocommerce/woocommerce-admin/pull/5825/commits/8e87f56328dcc5a502b17b426f05bda12f4a86d3